sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -21,15 +21,11 @@ std::unique_ptr<Response> CurlTransport::Send(Context& context, Request& request
     {
       case CURLE_COULDNT_RESOLVE_HOST:
       {
-        throw Azure::Core::Http::CouldNotResolveHostException();
-      }
-      case CURLE_WRITE_ERROR:
-      {
-        throw Azure::Core::Http::ErrorWhileWrittingResponse();
+        throw new Azure::Core::Http::CouldNotResolveHostException("Could not resolve host");
       }
       default:
       {
-        throw Azure::Core::Http::TransportException();
+        throw new Azure::Core::Http::TransportException("Error while sending request");
```

Comment:
Can you put any additional info in the message to help the customer diagnose & fix the issue?